sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -366,3 +344,86 @@ export class ServiceBusClient {
 function isReceiveMode(mode: any): mode is "peekLock" | "receiveAndDelete" {
   return mode && typeof mode === "string" && (mode === "peekLock" || mode === "receiveAndDelete");
 }
+
+/**
+ * Helper to validate and extract the common arguments from both the get*Receiver() overloads that
+ * have this pattern:
+ *
+ * queue, lockmode, options
+ * topic, subscription, lockmode, options
+ *
+ * @internal
+ * @ignore
+ */
+export function extractReceiverArguments<OptionsT>(
+  connectionStringEntityName: string | undefined,
+  queueOrTopicName1: string,
+  receiveModeOrSubscriptionName2: "peekLock" | "receiveAndDelete" | string,
+  receiveModeOrOptions3: "peekLock" | "receiveAndDelete" | OptionsT,
+  definitelyOptions4: OptionsT
+): {
+  entityPath: string;
+  receiveMode: "peekLock" | "receiveAndDelete";
+  options?: OptionsT;
+} {
+  if (isReceiveMode(receiveModeOrOptions3)) {
+    const topic = queueOrTopicName1;
+    const subscription = receiveModeOrSubscriptionName2;
+
+    validateEntityNamesMatch(connectionStringEntityName, topic, "receiver-topic");
+
+    return {
+      entityPath: `${topic}/Subscriptions/${subscription}`,
+      receiveMode: receiveModeOrOptions3,
+      options: definitelyOptions4
+    };
+  } else if (isReceiveMode(receiveModeOrSubscriptionName2)) {
+    validateEntityNamesMatch(connectionStringEntityName, queueOrTopicName1, "receiver-queue");
+
+    return {
+      entityPath: queueOrTopicName1,
+      receiveMode: receiveModeOrSubscriptionName2,
+      options: receiveModeOrOptions3
+    };
+  } else {
+    throw new TypeError("Invalid receiveMode provided");
```

Comment:
Would it be helpful to also echo what receiveMode they provided?

Reply:
It would be! :)

The only issue in this particular function is that the receive mode is how we figure out which overload we're in. So we only get into this 'else' if we couldn't find one altogether so we'd just be dumping two random parameters. :|